src/auth/auth-store.tsx

```diff
@@ -35,6 +37,22 @@ interface AuthState {
 export const useAuthStore = create<AuthState>()((set, get) => {
 	supabaseClient.auth.getSession().then(({ data: { session } }) => {
 		set({ session });
+
+		supabaseClient
+			.from("profiles")
+			.select("username")
+			.eq("id", get().session?.user?.id)
+			.then(({ data, error }) => {
+				if (error) throw new Error("Benutzername konnte nicht gefunden werden");
+
+				if (data.length > 1)
+					throw new Error(
+						"Benutzername konnte nicht eindeutig identifiziert werden",
+					);
+
+				const currentUsername = data[0].username;
+				set({ username: currentUsername });
+			});
```

Comment:
@aeschi could you check if replacing all this with `get().refreshUsername()` works? Would lead to less code duplicate/less lines in an already very big file (maybe we should think about splitting this file up at some point?).

Reply:
good idea! It works  🎉   
Yes, I agree it is already a very big file.